// The file defines protos used for CLgen telemetry data.
//
// Copyright (c) 2016, 2017, 2018, 2019 Chris Cummins.
//
// clgen is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// clgen is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with clgen.  If not, see <https://www.gnu.org/licenses/>.

syntax = "proto2";
package clgen;

// Records telemetry data about a single epoch of model training.
message ModelEpochTelemetry {
  // The number of milliseconds since the epoch that epoch training completed.
  optional int64 timestamp_utc_epoch_ms = 1;
  // The epoch which has just finished training, starting at one.
  optional int32 epoch_num = 2;
  // The wall time that it took to train the epoch.
  optional int64 epoch_wall_time_ms = 3;
  // The model's loss.
  optional double loss = 4;
}
